class/actions_legalnotice.class.php

```diff
@@ -94,11 +94,15 @@ function beforePDFCreation($parameters, &$object, &$action, $hookmanager)
 			$legal = new LegalNotice($this->db);
 			$TLegalNotice = $legal->fetchAll();
 
+			$TCountryUE = $this->search_country_EU();
+			$TCountryOutUE = $this->search_country_Out_EU();
+
 			foreach($TLegalNotice as &$legalNotice)
 			{
 				if ($object->thirdparty->tva_assuj !=
                     $legalNotice->is_assuj_tva && $legalNotice->is_assuj_tva != 2) continue;
-				if (!in_array($object->thirdparty->country_id, $legalNotice->fk_country) && !in_array(-1, $legalNotice->fk_country)) continue;
+				if (!in_array($object->thirdparty->country_id, $TCountryUE) && !in_array($object->thirdparty->country_id, $legalNotice->fk_country) && !in_array(-1, $legalNotice->fk_country) && !in_array(-3, $legalNotice->fk_country)) continue;
+				if (!in_array($object->thirdparty->country_id, $TCountryOutUE) && !in_array($object->thirdparty->country_id, $legalNotice->fk_country) && !in_array(-1, $legalNotice->fk_country)&& !in_array(-2, $legalNotice->fk_country)) continue;
```

Comment:
commente la condition pour expliquer ce que font tes ifs car pas ultra clair à lire
Sinon tu peux essayer de refacto avec un truc du genre ça à bien retester/renommer les variables pour que ça soit plus lisible
```suggestion
$isInCountryUE = in_array($object->thirdparty->country_id, $TCountryUE);
$isInCountryOutUE = in_array($object->thirdparty->country_id, $TCountryOutUE);
$isInLegalNoticeCountry = in_array($object->thirdparty->country_id, $legalNotice->fk_country);
$isInLegalNoticeCountryMinus1 = in_array(-1, $legalNotice->fk_country);
$isInLegalNoticeCountryMinus2 = in_array(-2, $legalNotice->fk_country);
$isInLegalNoticeCountryMinus3 = in_array(-3, $legalNotice->fk_country);

if ((!$isInCountryUE || !$isInLegalNoticeCountry || !$isInLegalNoticeCountryMinus1 || !$isInLegalNoticeCountryMinus3) && (!$isInCountryOutUE || !$isInLegalNoticeCountry || !$isInLegalNoticeCountryMinus1 || !$isInLegalNoticeCountryMinus2)) {
    continue;
}
```